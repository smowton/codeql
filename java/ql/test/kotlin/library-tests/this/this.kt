
class Outer {
    inner class Inner {
        fun someFun() {
            val x1: Inner = this
            val x2: Inner = this@Inner
            val x3: Outer = this@Outer

            val labelledExtensionFun = someLabelledExtensionFun@ fun ExtensionClass.() {
                val x4: ExtensionClass = this
                val x5: ExtensionClass = this@someLabelledExtensionFun
            }

            val unLabelledExtensionFun = fun ExtensionClass.() {
                val x6: ExtensionClass = this
            }

            val someLambda = { i: Int ->
                val x7: Inner = this
            }
        }

        fun ExtensionClass.extensionFun() {
            val x1: ExtensionClass = this
            val x2: Inner = this@Inner
            val x3: Outer = this@Outer
            val x4: ExtensionClass = this@extensionFun
        }

        fun innerCaller() {
            topLevelFun()
            outerFun()
            innerFun()
            topLevelOuterFun()
            topLevelInnerFun()
            outerInnerFun()
            topLevelOuterInnerFun()
            this.innerFun()
            this.topLevelInnerFun()
            this.outerInnerFun()
            this.topLevelOuterInnerFun()
        }

        fun innerFun() {}
        fun topLevelInnerFun() {}
        fun outerInnerFun() {}
        fun topLevelOuterInnerFun() {}
    }
    fun outerFun() {}
    fun topLevelOuterFun() {}
    fun outerInnerFun() {}
    fun topLevelOuterInnerFun() {}
}
fun topLevelFun() {}
fun topLevelOuterFun() {}
fun topLevelInnerFun() {}
fun topLevelOuterInnerFun() {}

class ExtensionClass {
}
