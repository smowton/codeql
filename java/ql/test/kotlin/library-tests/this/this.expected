| this.kt:5:29:5:32 | this | file://:0:0:0:0 | <none> |
| this.kt:6:29:6:38 | this | file://:0:0:0:0 | <none> |
| this.kt:7:29:7:38 | this | file://:0:0:0:0 | <none> |
| this.kt:29:38:29:41 | this | file://:0:0:0:0 | <none> |
| this.kt:30:29:30:38 | this | file://:0:0:0:0 | <none> |
| this.kt:31:29:31:38 | this | file://:0:0:0:0 | <none> |
| this.kt:32:38:32:54 | this | file://:0:0:0:0 | <none> |
| this.kt:37:13:37:22 | this | file://:0:0:0:0 | <none> |
| this.kt:38:13:38:22 | this | file://:0:0:0:0 | <none> |
| this.kt:39:13:39:30 | this | file://:0:0:0:0 | <none> |
| this.kt:40:13:40:30 | this | file://:0:0:0:0 | <none> |
| this.kt:41:13:41:27 | this | file://:0:0:0:0 | <none> |
| this.kt:42:13:42:35 | this | file://:0:0:0:0 | <none> |
| this.kt:43:13:43:16 | this | file://:0:0:0:0 | <none> |
| this.kt:44:13:44:16 | this | file://:0:0:0:0 | <none> |
| this.kt:45:13:45:16 | this | file://:0:0:0:0 | <none> |
| this.kt:46:13:46:16 | this | file://:0:0:0:0 | <none> |
